tests/late_joiners.py

```diff
@@ -36,117 +35,92 @@ def timeout_handler(node, suspend, election_timeout):
         node.resume()
 
 
-def find_primary(network):
-    # track if there is a new primary
-    term_info = {}
+def update_term_info(network, term_info):
     try:
         cur_primary, cur_term = network.find_primary()
         term_info[cur_term] = cur_primary.node_id
     except TimeoutError:
         LOG.info("Trying to access a suspended network")
-    return term_info
 
 
-def get_node_local_commit(node):
-    with node.node_client() as c:
-        r = c.get("getLocalCommit")
-        return r.commit, r.global_commit
+@reqs.description("Running transactions against logging app")
+@reqs.supports_methods("LOG_record", "LOG_record_pub", "LOG_get", "LOG_get_pub")
+@reqs.at_least_n_nodes(3)
+def test_run_txs(
+    network,
+    args,
+    nodes=None,
+    num_txs=1,
+    start_idx=0,
+    timeout=3,
+    can_fail=False,
+    notifications_queue=None,
+    verify=True,
+    wait_for_sync=False,
+):
+    txs = app.LoggingTxs(
+        notifications_queue=notifications_queue,
+        tx_index_start=start_idx,
+        can_fail=can_fail,
+        timeout=timeout,
+        wait_for_sync=wait_for_sync,
+    )
+    if nodes is None:
+        nodes = network.get_joined_nodes()
+    num_nodes = len(nodes)
+
+    for tx in range(num_txs):
+        txs.issue_on_node(
+            network=network,
+            remote_node=nodes[tx % num_nodes],
+            number_txs=1,
+            consensus=args.consensus,
+        )
 
+    if verify:
+        txs.verify_last_tx(network)
+    else:
+        LOG.warning("Skipping log messages verification")
 
-def wait_for_late_joiner(old_node, late_joiner):
-    old_node_lc, old_node_gc = get_node_local_commit(old_node)
-    LOG.success(
-        f"node {old_node.node_id} is at state local_commit:{old_node_lc}, global_commit:{old_node_gc}"
-    )
-    while True:
-        lc, gc = get_node_local_commit(late_joiner)
-        LOG.success(
-            f"late joiner {late_joiner.node_id} is at state local_commit:{lc}, global_commit:{gc}"
+    return network
+
+
+@reqs.description("Adding a very late joiner")
+def test_add_late_joiner(network, args, nodes_to_keep):
+    new_node = network.create_and_trust_node(args.package, "localhost", args)
+    nodes_to_keep.append(new_node)
+    return network
+
+
+@reqs.description("Suspend nodes")
+@reqs.at_least_n_nodes(3)
+def test_suspend_nodes(network, args, nodes_to_keep):
```

Comment:
It would be awesome to add this to our existing test suite. We could check for elections without having to kill one node (i.e. just suspend the primary for long enough).